project-base/storefront/hooks/useQueryParams.ts

```diff
@@ -273,12 +280,22 @@ export const useQueryParams = () => {
     const pushQueryPage = (page: number) => {
         const newQuery: UrlQueries = {
             ...query,
+            [LOAD_MORE_QUERY_PARAMETER_NAME]: undefined,
             [PAGE_QUERY_PARAMETER_NAME]: page > 1 ? page.toString() : undefined,
         } as const;
 
         pushQueries(newQuery, true);
     };
 
+    const pushQueryLoadMore = (loadMore: number) => {
+        const newQuery: UrlQueries = {
+            ...query,
+            [LOAD_MORE_QUERY_PARAMETER_NAME]: loadMore > 0 ? loadMore.toString() : undefined,
+        } as const;
+
+        pushQueries(newQuery, true);
+    };
+
```

Comment:
Since this is used only on one place cannot we use it directly there? I don't see that this would improve anything, just splitting the code to another place.

```suggestion
const loadMore = () => {
        const newQuery: UrlQueries = {
            ...query,
            [LOAD_MORE_QUERY_PARAMETER_NAME]: loadMore > 0 ? loadMore.toString() : undefined,
        } as const;

        pushQueries(newQuery, true);
    };
```

Reply:
You're right, let's do it as you say.